packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -580,67 +582,71 @@ class TimelineFrame {
   /// Whether the frame is ready for the timeline.
   ///
   /// A frame is ready once it has both required event flows as well as
-  /// [startTime] and [endTime].
+  /// [_pipelineItemStartTime] and [_pipelineItemEndTime].
   bool get isReadyForTimeline {
     return uiEventFlow != null &&
         gpuEventFlow != null &&
-        _startTime != null &&
-        _endTime != null;
+        _pipelineItemStartTime != null &&
+        _pipelineItemEndTime != null;
   }
 
-  /// Frame start time in micros.
+  /// Pipeline item start time in micros.
   ///
-  /// We take the min of [uiStartTime] and [_startTime] because we use an
-  /// epsilon when determining if an event fits within frame boundaries.
-  /// Therefore, there is a chance that [uiStartTime] could be less than
-  /// [_startTime].
-  int get startTime => nullSafeMin(_startTime, uiStartTime);
-  int _startTime;
-  set startTime(int time) => _startTime = nullSafeMin(_startTime, time);
-
-  /// Frame end time in micros.
+  /// This stores the timestamp of the start pipeline event for this frame. We
+  /// use this value to determine whether a TimelineEvent fits within the
+  /// frame's time boundaries.
+  int get pipelineItemStartTime => _pipelineItemStartTime;
+  int _pipelineItemStartTime;
+  set pipelineItemStartTime(int time) =>
+      _pipelineItemStartTime = nullSafeMin(_pipelineItemStartTime, time);
+
+  /// Pipeline item end time in micros.
   ///
-  /// We take the max of [gpuEndTime] and [_endTime] because we use an epsilon
-  /// when determining if an event fits within frame boundaries. Therefore,
-  /// there is a chance that [gpuEndTime] could be greater than [_endTime].
-  int get endTime => nullSafeMax(_endTime, gpuEndTime);
-  int _endTime;
-  set endTime(int time) => _endTime = nullSafeMax(_endTime, time);
+  /// This stores the timestamp of the end pipeline event for this frame. We use
+  /// this value to determine whether a TimelineEvent fits within the frame's
+  /// time boundaries.
+  int get pipelineItemEndTime => _pipelineItemEndTime;
+  int _pipelineItemEndTime;
+  set pipelineItemEndTime(int time) =>
+      _pipelineItemEndTime = nullSafeMax(_pipelineItemEndTime, time);
+
+  bool get isWellFormed =>
+      _pipelineItemStartTime != null && _pipelineItemEndTime != null;
 
-  bool get isWellFormed => _startTime != null && _endTime != null;
+  /// Frame start time in micros.
+  int get startTime => _uiStartTime;
```

Comment:
there are getting to be a lot of startTime end time pairs here. 
I would create
a
TimeRange class to standardize this.